tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py

```diff
@@ -2109,7 +2109,7 @@ def test_trainable_coeffs(self, broadcast, tol):
 
         res = dev.execute([tape])
         expected = -c * np.sin(x) * np.sin(y) + np.cos(x) * (a + b * np.sin(y))
-        assert np.allclose(res, expected, atol=tol, rtol=0)
+        assert np.allclose(res, expected, atol=tol, rtol=tol)
```

Comment:
```suggestion
        # This test is mostly for the shot vector pipeline check
        # therefore only check a rough accuracy
        assert res.shape == len(shot_vec)
        assert np.allclose(res, expected, atol=0.002, rtol=0)
```
@andrijapau @astralcai How about this? This absorbs both suggestions: 1. we only set `atol` now, more straightforward 2. checks the shape match based on the given `shot_vec` 2. short comment on decision.

Reply:
LGTM

Reply:
Note: I changed `shape(res[0])` -> `shape(res)[1]`

Reply:
Looks good to me, why is it index [1] and [0] shouldn't they be the same?

Reply:
@andrijapau the actual `res` is of shape `(1, 3)`, super weird.